src/paho/mqtt/client.py

```diff
@@ -2137,13 +2138,13 @@ def _message_retry_check_actual(self, messages, mutex):
         mutex.release()
 
     def _message_retry_check(self):
-        self._message_retry_check_actual(self._out_messages, self._out_message_mutex)
-        self._message_retry_check_actual(self._in_messages, self._in_message_mutex)
+        self._message_retry_check_actual(self._out_messages.values(), self._out_message_mutex)
```

Comment:
accessing to self._out_messages and self._in_messages .values() must be done inside _message_retry_check_actual after locking the mutex